python/hello-world-gpu/readme.md

```diff
@@ -0,0 +1,274 @@
+sudo systemctl restart containerd
+```
+
+Running a Sample Workload with Docker:
+
+```bash
+sudo docker run --rm --runtime=nvidia --gpus all ubuntu nvidia-smi
+```
+
+Your output should resemble the following output:
+
+```bash
++-----------------------------------------------------------------------------+
+| NVIDIA-SMI 535.86.10    Driver Version: 535.86.10    CUDA Version: 12.2     |
+|-------------------------------+----------------------+----------------------+
+| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
+| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
+|                               |                      |               MIG M. |
+|===============================+======================+======================|
+|   0  Tesla T4            On   | 00000000:00:1E.0 Off |                    0 |
+| N/A   34C    P8     9W /  70W |      0MiB / 15109MiB |      0%      Default |
+|                               |                      |                  N/A |
++-------------------------------+----------------------+----------------------+
+
++-----------------------------------------------------------------------------+
+| Processes:                                                                  |
+|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
+|        ID   ID                                                   Usage      |
+|=============================================================================|
+|  No running processes found                                                 |
++-----------------------------------------------------------------------------+
+```
+If you run into the following issue:
+```bash
+> docker run --rm --gpus all ubuntu nvidia-smi
+Failed to initialize NVML: GPU access blocked by the operating system Failed to properly shut down NVML: GPU access blocked by the operating system
+```
+
+Inside the file /etc/nvidia-container-runtime/config.toml change no-cgroups from true to false.
+
+Configure the nvidia-container-runtime as the default low-level runtime to use:
+
+Edit Docker Daemon Configuration:
+
+Open the Docker daemon configuration file using a text editor. This file is typically located at /etc/docker/daemon.json.
+
+```bash
+sudo nano /etc/docker/daemon.json
+```
+
+Configure the Default Runtime:
+
+Add or update the default-runtime key to specify nvidia as the default runtime. Ensure the file includes a section for runtimes and sets nvidia-container-runtime as the path for the nvidia runtime. Your daemon.json should look something like this:
+
+```bash
+{
+  "runtimes": {
+    "nvidia": {
+      "path": "nvidia-container-runtime",
+      "runtimeArgs": []
+    }
+  },
+  "default-runtime": "nvidia"
+}
+```
+
+Restart Docker Daemon:
+
+After saving the changes, restart the Docker daemon to apply the new configuration.
+
+```bash
+sudo systemctl restart docker
+```
+
+### 6. NVIDIA device plugin for Kubernetes
+
```

Comment:
This part should be in the terraform deployment I think.

Reply:
This is related to the deployment of Kubernetes itself, not ArmoniK. So it makes sense that is it a separated step.

Reply:
if deploying in local, one should add the plugin to the cluster. In the cloud this step ain't needed.